pulsar-client-1x-base/pulsar-client-2x-shaded/pom.xml

```diff
@@ -0,0 +1,91 @@
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.apache.pulsar</groupId>
+    <artifactId>pulsar-client-1x-base</artifactId>
+    <version>2.3.0-SNAPSHOT</version>
+    <relativePath>..</relativePath>
+  </parent>
+
+  <artifactId>pulsar-client-2x-shaded</artifactId>
```

Comment:
this is confusing. `pulsar-client-1x-base` has child module `pulsar-client-2x-shaded` ? client-2.x already has separate module at parent level. Shouldn't we need `pulsar-client-1x-shaded` ?

Reply:
The problem I'm trying to solve here is that `pulsar-client-1x` is a wrapper over `pulsar-client`. At the same time, we have some interfaces that need to be duplicated in `pulsar-client-1x` (because the 2.x version will be different). 

So, `pulsar-client-2x-shaded` is really `pulsar-client` but with just few interfaces renamed (eg: `PulsarClient`, `Producer`, `Consumer` and `Reader`) to avoid conflicts with the 1.x API. 

So, although confusing, `pulsar-client-2x-shaded` is really the correct name and this module is not meant for direct usage anyway.